pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +194,154 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def get_transformation_block(
+    in_channels: int,
+    kernel_size: Union[Tuple[int, int], int],
+    stride: Union[Tuple[int, int], int],
+    padding: Union[Tuple[int, int], int],
+    impl_params: bool = True,
+) -> nn.Module:
+    if impl_params:
+        # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/module.py#L246
+        return nn.AvgPool2d(kernel_size=kernel_size, stride=stride, padding=padding)
+    return cast(
+        nn.Module,
+        nn.utils.weight_norm(
+            nn.Conv2d(in_channels, 3, kernel_size, stride=stride, padding=padding)
+        ),
+    )
+
+
+class TransformerBlock(nn.Module):
```

Comment:
Just realized this: the transformer block is also only needed for the loss calculation, correct? If so, please remove this for now to keep the scope of PR tight.

Reply:
In this case, the `DiscriminatorMultiLayerEncoder` is also only required for the calculation of the loss. Therefore I would suggest at this point that I extend this PR by the necessary classes and function from loss?

Reply:
This raises the mental burden to review this, which I like to avoid. Please remove the out-of-scope parts for now and re-add them in a follow-up PR.